### PR TITLE
planner: separate cost model ver1/ver2 into different files (part 1)

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #36243

Problem Summary:

### What is changed and how it works?

just refactor, separate cost model ver1/ver2 into different files

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/core/plan_cost_ver2.go [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+	// consider tidb_enforce_mpp
+	_, isMPP := p.tablePlan.(*PhysicalExchangeSender)
+	if isMPP && p.ctx.GetSessionVars().IsMPPEnforced() &&
+		!hasCostFlag(option.CostFlag, CostFlagRecalculate) { // show the real cost in explain-statements
+		p.planCost /= 1000000000
+	}
+	return p.planCost, nil
+}
+
+// plan-cost = index-side-cost + (table-side-cost + double-read-cost) / double-read-concurrency
```
Comment (winoros) [inline planner/core/plan_cost_ver2.go:144]: We'd better follow the format that the first comment begins with `funcName ...`

Comment (qw4990) [reply planner/core/plan_cost_ver2.go:151]: Updated, PTAL.

File: planner/core/plan_cost_test.go [hunk 2]
```diff
@@ -119,12 +120,13 @@ func TestScanOnSmallTable(t *testing.T) {
 	}
 
 	rs := tk.MustQuery("explain select * from t").Rows()
-	useTiKVScan := true
+	useTiKVScan := false
 	for _, r := range rs {
+		fmt.Println(">>> ", r)
```
Comment (Reminiscent) [inline planner/core/plan_cost_test.go:125]: Remove this part.
BTW, why we change the set order for `useTiKVScan`.

File: planner/core/plan_cost.go [hunk 3]
```diff
@@ -509,14 +477,7 @@ func (p *PhysicalIndexScan) GetPlanCost(_ property.TaskType, option *PlanCostOpt
...
-			scanFactor = p.ctx.GetSessionVars().GetDescScanFactor(p.Table)
-		}
-		rowCount = getCardinality(p, costFlag)
-		rowSize = math.Max(p.getScanRowSize(), 2.0)
-		logRowSize := math.Log2(rowSize)
-		selfCost = rowCount * logRowSize * scanFactor
+		return p.getPlanCostVer2(taskType, option)
 	}
 	if option.tracer != nil {
 		setPhysicalTableOrIndexScanCostDetail(p, option.tracer, rowCount, rowSize, scanFactor, costModelVersion)
```
Comment (Reminiscent) [inline planner/core/plan_cost.go:483]: It seems we not use the trace for the version2.

**Review by Reminiscent — APPROVED** [review]

File: planner/core/plan_cost_test.go (hunk 2 above)
Comment (qw4990) [reply planner/core/plan_cost_test.go]: My mistake. It's removed.

File: planner/core/plan_cost.go (hunk 3 above)
Comment (qw4990) [reply planner/core/plan_cost.go:483]: Yes. I'll support it for ver2 later in the next a few PRs.

**Review by winoros — APPROVED** [review]

**Review by Yisaer — APPROVED** [review]

**Review by time-and-fate — APPROVED** [review]

**Pull request merged**